### ISSUE TITLE
Add support for mypy

### ISSUE THREAD

**Issue by Hnasar** [Bachmann1234/diff-cover#74]
It looks like `mypy` has support for outputting Cobertura reports:
https://github.com/python/mypy/blob/master/mypy/report.py#L481

Maybe it'd be trivial to add support to `mypy` for `diff-cover` and `diff-quality`?

**Comment by Bachmann1234** [comment]
If Mypy outputs Cobertura then you can already use diff-cover! You just use that mypy coverage file. In theory I think it would all just work

**Comment by blueyed** [comment]
Yeah, it should work already.  Working on this for pytest in https://github.com/pytest-dev/pytest/pull/6486.
This issue can be closed.

**Comment by blueyed** [comment]
Maybe adding support for `--violations=mypy` might make sense though (https://github.com/Bachmann1234/diff_cover#adding-diff-quality-support-for-a-new-quality-checker), but that would have to be done for mypy itself then.

**Comment by m-aciek** [comment]
Mypy's [Cobertura report](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-cobertura-xml-report) is lackonic. It treats lines that generate warning or error as "uncovered", but doesn't provide a cause. The sample output from a run is:
```zsh
% mypy … --cobertura-xml-report .
[all violations printed]
Generated Cobertura report: /Users/maciej.olko/projects/all-the-things/cobertura.xml
Found 38 errors in 1 file (checked 1 source file)
% uvx diff-cover cobertura.xml --compare-branch master
-------------
Diff Coverage
Diff: master...HEAD, staged and unstaged changes
-------------
[…].py (50.0%): Missing lines 101,108-109,140,172,175
-------------
Total:   12 lines
Missing: 6 lines
Coverage: 50%
-------------
```

It's definitely worth to integrate mypy, that would allow to see the violations content.

**Comment by YeeaaahMan** [comment]
I've prepared a PR here: Bachmann1234/diff_cover/pull/540
